src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -557,21 +557,10 @@ private static FormulaValue OperatorDynamicGetField(BinaryOpNode node, FormulaVa
 
             if (arg1 is UntypedObjectValue cov && arg2 is StringValue sv)
             {
-                if (cov.Impl.Type is ExternalType et && (et.Kind == ExternalTypeKind.Object || et.Kind == ExternalTypeKind.ArrayAndObject))
-                {
-                    if (cov.Impl.TryGetProperty(sv.Value, out var res))
-                    {
-                        if (res.Type == FormulaType.Blank)
-                        {
-                            return new BlankValue(node.IRContext);
-                        }
-
-                        return new UntypedObjectValue(node.IRContext, res);
-                    }
-                    else
-                    {
-                        return new BlankValue(node.IRContext);
-                    }
+                if (cov.Impl is UntypedObjectBase untypedObjectBase && cov.Impl.Type is ExternalType et && (et.Kind == ExternalTypeKind.Object || et.Kind == ExternalTypeKind.ArrayAndObject))
+                {
+                    // Passing Span in case host wants to return an error.
+                    return untypedObjectBase.GetProperty(sv.Value, node.IRContext.ResultType, node.IRContext.SourceContext);
```

Comment:
>node.IRContext.SourceContext [](http://example.com/codeflow?start=94&length=28)

I don't think we have any meaningful span here. 
(we should just remove IRContext.SourceContext) 

I'd just remove this param.